lib/pbench/server/api/resources/upload_api.py

```diff
@@ -51,69 +51,75 @@ def __init__(self, config, logger):
 
     @Auth.token_auth.login_required()
     def put(self, controller):
-
         try:
             username = Auth.token_auth.current_user().username
         except Exception:
             self.logger.exception("Tarfile upload: Exception verifying the username")
             abort(HTTPStatus.INTERNAL_SERVER_ERROR, message="INTERNAL ERROR")
 
-        if not request.headers.get("filename"):
-            self.logger.debug(
-                "Tarfile upload: Post operation failed due to missing filename header"
+        filename_hdr = request.headers.get("filename")
+        if not filename_hdr:
+            self.logger.info(
```

Comment:
This needs to be resolved between your PR and Nikhil's. At one point, we'd more or less decided to reserve `warning` log level for "user errors", and this qualifies. While I was never 100% comfortable with that convention, it seems at least plausible without creating our own additional log level. Whereas mixing internal informational status messages with user errors makes the log harder to parse. All in all, I think the `warning` convention is the best idea we'd seen in this area... but, more pragmatically, having you change it to `info` and then Nikhil immediately changing it to `warning` doesn't help anyone. ;-)

Reply:
Yes, good point.  Fixed.